packages/dds/merge-tree/src/referencePositions.ts

```diff
@@ -56,12 +56,35 @@ export function refHasRangeLabels(refPos: ReferencePosition): boolean {
     return refGetRangeLabels(refPos) !== undefined;
 }
 
+/**
+ * Represents a reference to a place within a merge tree. This place conceptually remains stable over time
+ * by referring to a particular segment and offset within that segment.
+ * Thus, this reference's character position changes as the tree is edited.
+ */
 export interface ReferencePosition {
+    /**
+     * @returns - Properties associated with this reference
+     */
     properties?: PropertySet;
     refType: ReferenceType;
 
+    /**
+     * Gets the segment that this reference position is semantically associated with. Returns undefined iff the
+     * reference became detached from the string.
+     */
     getSegment(): ISegment | undefined;
+
+    /**
+     * Gets the offset for this reference position relative to the segment it's associated to.
```

Comment:
This is slightly confusing to me - is it the offset from the root relative to this reference position's segment? I may also just not have a great understanding of offset, so feel free to disregard this comment if that is the case.

Reply:
nah, it's good feedback. I was motivated to doc this interface after getting a question from someone trying to use it. We want it to be abundantly clear that this is *not* an absolute position within the merge-tree. I added an example to demonstrate that.